### PR TITLE
Grid Select Improvements

### PR DESCRIPTION
**Pull request by stdavis**
- [x] Remove "NaN" values
- [x] Fix "[other]" bug
- [x] Other "other" dialog improvements

You'll just have to believe me that the tests are passing.

### CODE REVIEW

File: _src/app/Domains.js [hunk 1]
```diff
@@ -67,29 +65,20 @@ define([
...
-
-            array.forEach(select.children, function (option) {
-                if (option.value !== '' && option.value !== this.otherTxt) {
-                    existingOptions.push({
-                        code: option.value,
-                        name: option.innerHTML
-                    });
-                }
-            }, this);
+            var existingOptions = Array.from(select.children).map(option => option.value);
```
Comment (steveoh) [inline _src/app/Domains.js]: This might need to be polyfilled if people are using IE at all.

Comment (stdavis) [reply _src/app/Domains.js]: Ah, I thought that Babel would take care of this one for me. I'm not too worried about supporting IE at this point for this project.

File: _src/app/_GridMixin.js [hunk 2]
```diff
@@ -32,6 +40,16 @@ define([
...
     //      Mixin to add dgrid to a widget.
     return declare(null, {
+        // this is to remove the default NaN value if there is an empty string
+        // it's used as a column editor in classes that use this mixin
+        NewNumberSpinner: declare([NumberSpinner], {
+            value: null,
+            _getValueAttr() {
+                const inheritedValue = this.inherited(arguments);
+
+                return isNaN(inheritedValue) ? null : inheritedValue;
```
[the remark this answers is not in the document]
Comment (stdavis) [reply _src/app/_GridMixin.js]: No. In fact, it fixes problems such as attempting to write a string (`NaN`) to numeric fields.

[the remark this answers is not in the document]
Comment (steveoh) [reply _src/app/_GridMixin.js]: Does 0 get written to the db?

**Pull request merged**